hockeypuck_rock/delete_keys.py

```
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Delete keys from the Hockeypuck Postgres database by fingerprint."""
+
+import argparse
+import logging
+import os
+import re
+import sys
+from typing import List
+
+import psycopg2
+
+logger = logging.getLogger(__name__)
+
+
+def get_db_connection() -> psycopg2.extensions.connection | None:
```

I'd rather re-raise the exception rather than returning a None value since it would be an abnormal flow

---

addressed in the next [PR](https://github.com/canonical/hockeypuck-k8s-operator/pull/12)